graphql-dgs-codegen-core/src/test/kotlin/com/netflix/graphql/dgs/codegen/CodeGenTest.kt

```diff
@@ -550,6 +550,39 @@ class CodeGenTest {
         assertCompiles(dataTypes)
     }
 
+    @Test
+    fun generateInputWithDefaultValueForEnum() {
+        val schema = """
+            enum Color {
+                red
+            }
+            
+            input ColorFilter {
+                color: Color = red
+            }
+        """.trimIndent()
+
+        val (dataTypes) = CodeGen(CodeGenConfig(schemas = setOf(schema), packageName = basePackageName)).generate() as CodeGenResult
+        assertThat(dataTypes).hasSize(1)
+
+        val data = dataTypes[0]
+        assertThat(data.packageName).isEqualTo(typesPackageName)
+
+        val type = data.typeSpec
+        assertThat(type.name).isEqualTo("ColorFilter")
+
+        val fields = type.fieldSpecs
+        assertThat(fields).hasSize(1)
+
+        val colorField = fields[0]
+        assertThat(colorField.name).isEqualTo("color")
+        assertThat(colorField.type.toString()).isEqualTo("$typesPackageName.Color")
+        assertThat(colorField.initializer.toString()).isEqualTo("$typesPackageName.Color.red")
+
+        // This does not work as the compiler cannot find the generated enum and fails to resolve the type of the field.
+        // assertCompiles(dataTypes)
```

Comment:
Did you try passing in the enumTypes from CodeGenResult as well? That should work.

Reply:
Yes - you just need enumTypes in addition to dataTypes

Reply:
Thanks ... that works.

Fixed